### PR TITLE
Change default interpolation to nearest

### PR DESCRIPTION
**Pull request by jenshnielsen**
As discussed in the Scipy BOF I propose changing the default interpolation of imshow to nearest. 

I haven't yet refreshed any of the tests. So this will likely fail.

### CODE REVIEW

**Comment by jenshnielsen** [comment]
matplotlib.tests.test_axes.test_specgram_angle_freqs.test
matplotlib.tests.test_axes.test_specgram_magnitude_noise.test
matplotlib.tests.test_axes.test_specgram_magnitude_noise.test
matplotlib.tests.test_axes.test_specgram_noise.test
matplotlib.tests.test_axes.test_specgram_noise.test
matplotlib.tests.test_axes.test_specgram_noise_angle.test
matplotlib.tests.test_backend_svg.test_noscale.test
matplotlib.tests.test_backend_svg.test_noscale.test
matplotlib.tests.test_image.test_image_clip.test
matplotlib.tests.test_image.test_image_clip.test
matplotlib.tests.test_image.test_image_clip.test
matplotlib.tests.test_image.test_image_cliprect.test
matplotlib.tests.test_image.test_image_cliprect.test
matplotlib.tests.test_image.test_image_cliprect.test
matplotlib.tests.test_image.test_image_composite_background.test
matplotlib.tests.test_image.test_image_composite_background.test
matplotlib.tests.test_image.test_image_composite_background.test
matplotlib.tests.test_image.test_rasterize_dpi.test
matplotlib.tests.test_image.test_rasterize_dpi.test
matplotlib.tests.test_patheffects.test_patheffect1.test
matplotlib.tests.test_patheffects.test_patheffect1.test
matplotlib.tests.test_patheffects.test_patheffect1.test
matplotlib.tests.test_patheffects.test_patheffect2.test
matplotlib.tests.test_patheffects.test_patheffect2.test matplotlib.tests.test_patheffects.test_patheffect2.test mpl_toolkits.tests.test_axes_grid.test_imagegrid_cbar_mode_edge.test

**Comment by efiring** [comment]
Changing to "nearest" makes sense to me; I have never seen the present default as useful.

**Comment by tacaswell** [comment]
:+1:

File: lib/matplotlib/image.py [hunk 1]
```diff
@@ -714,8 +714,8 @@ class NonUniformImage(AxesImage):
     def __init__(self, ax, **kwargs):
         """
         kwargs are identical to those for AxesImage, except
-        that 'interpolation' defaults to 'nearest', and 'bilinear'
-        is the only alternative.
+        that 'nearest', and 'bilinear' are the only supported 'interpolation'
```
Comment (maxalbert) [inline lib/matplotlib/image.py]: Should the comma be removed between "nearest" and "and 'bilinear'"?

**Comment by mdboom** [comment]
I agree with this change.  However, the one downside is that it will increase file sizes for the vector backends (since the data will now be upsampled to the output resolution).  We should probably include that as a note in the "What's new" about this change.

**Comment by WeatherGod** [comment]
Doesn't that depend on whether the default is 'none' versus 'nearest'?

**Comment by mdboom** [comment]
@WeatherGod: You're right.  Ignore my comment.

File: lib/matplotlib/image.py (hunk 1 above)
Comment (endolith) [inline lib/matplotlib/image.py]: @maxalbert  yes

**Comment by astrofrog** [comment]
:+1: for changing to nearest by default

**Comment by jradavenport** [comment]
:+1: for nearest. I'm seeing more and more figures making it in to published literature using 'none', which are unreadable. Changing this default would promote best-practices for scientific figures.

**Comment by jenshnielsen** [comment]
replaced by #5416 against the correct branch and with the typo fixed

**Pull request closed without merging**